homeassistant/components/sensor/syncthru.py

```diff
@@ -0,0 +1,370 @@
+            devices.append(SyncThruInputTray(hass, printer, name, key))
+    for key in printer.outputTrayStatus().keys():
+        if 'output_tray_' + str(key) in monitored:
+            devices.append(SyncThruOutputTray(hass, printer, name, key))
+
+    add_devices(devices, True)
+    return True
+
+
+class SyncThruSensor(Entity):
+    """Implementation of a Samsung Printer sensor platform."""
+
+    def __init__(self, hass, syncthru, name):
+        """Initialize the sensor"""
+        self._hass = hass
+        self.syncthru = syncthru
+        # init the devices entitiy name starting without number/name
+        self._attributes = {}
+        self._state = STATE_UNKNOWN
+        self._name = name
+        self._icon = 'mdi:printer'
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Return the state of the device."""
+        return self._icon
+
+    def update(self):
+        """Get the latest data from SyncThru and update the state."""
+        syncthru = self.syncthru
+
+        syncthru.update()
+        self._state = syncthru.deviceStatus()
+
+        if syncthru.isOnline():
+            self._attributes[CONF_FRIENDLY_NAME] = syncthru.model()
+            self._state = syncthru.deviceStatus()
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        return self._attributes
+
+
+class SyncThruToner(Entity):
+    """Implementation of a Samsung Printer toner sensor platform."""
+
+    def __init__(self, hass, syncthru, name, color):
+        """Initialize the sensor"""
+        self._hass = hass
+        self.syncthru = syncthru
+        # init the devices entitiy name starting without number/name
+        self._attributes = {}
+        self._state = STATE_UNKNOWN
+        self._name = "{} toner {}".format(name, color)
+        self._color = color
+        self._icon = 'mdi:printer'
+        self._unit_of_measurement = '%'
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Return the state of the device."""
+        return self._icon
+
+    @property
+    def unit_of_measurement(self):
+        """Return the state of the device."""
+        return self._unit_of_measurement
+
+    def update(self):
+        """Get the latest data from SyncThru and update the state."""
+
+        # Data fetching is taken care of through the Main sensor
+        syncthru = self.syncthru
+
+        if syncthru.isOnline():
+            self._attributes = syncthru.tonerStatus(
+                ).get(self._color, {})
```

Comment:
Having the closing parenthesis on a beginning of the new line looks weird when there's no parameters listed at all, but I assume that's ok.